README.md

```diff
@@ -1,3 +1,262 @@
+provides a SQL-like interface for querying and manipulating data. This is done
+through [DataFrame and DataSet](https://spark.apache.org/docs/latest/sql-programming-guide.html),
+Spark SQL's data model, built on top of RDDs.
+
+### Supported Spark runtime environments and requirements
+
+You can use the connector with Spark locally, as well as in managed environments
+such as Dataproc cluster or serverless. You need the following depending on the
+environments you choose to use:
+
+| Runtime environment              | Bigtable | Dataproc | Cloud Storage |
+|----------------------------------|----------|----------|---------------|
+| Local Spark w/ Bigtable emulator | Optional | Optional | Optional      |
+| Local Spark                      | Required | Optional | Optional      |
+| Dataproc Cluster                 | Required | Required | Optional      |
+| Dataproc Serverless              | Required | Required | Required      |
+
+### Supported Spark versions
+
+Our connector supports the following Spark versions **with Scala 2.12**:
+
+| Scala version | Spark versions                    | Spark Application Languages                             |
+|---------------|-----------------------------------|---------------------------------------------------------|
+| 2.12          | 2.4.8, 3.1.x, 3.2.x, 3.4.x, 3.5.x | Java, Scala, PySpark (`.py` files or Jupyter notebooks) |
+
+## Main features
+
+For a detailed list of features and how to use them, you can refer the official
+documentation
+[here](https://cloud.google.com/bigtable/docs/use-bigtable-spark-connector).
+A list of main features is as follows:
+
+### Catalog definition
+
+You can define a catalog as a JSON-formatted string, to convert from the
+DataFrame's schema to a
+format compatible with Bigtable. This is an example of a catalog JSON:
+
+```
+{
+  "table": {"name": "t1"},
+  "rowkey": "id_rowkey",
+  "columns": {
+    "id": {"cf": "rowkey", "col": "id_rowkey", "type": "string"},
+    "name": {"cf": "info", "col": "name", "type": "string"},
+    "birthYear": {"cf": "info", "col": "birth_year", "type": "long"},
+    "address": {"cf": "location", "col": "address", "type": "string"}
+  }
+}
+```
+
+Here, the columns `name`, `birthYear`, and `address` from the DataFrame are
+converted into Bigtable
+columns and the `id` column is used as the row key. Note that you could also
+specify *compound* row keys,
+which are created by concatenating multiple DataFrame columns together.
+
+### Writing to Bigtable
+
+You can use the `dataFrame.write()` format write to Bigtable.
+This is a sample snippet of writing to Bigtable using Java:
+
+```java
+Dataset<Row> dataFrame;
+// Adding some values to dataFrame.
+dataFrame
+  .write()
+  .format("bigtable")
+  .option("catalog", catalog)
+  .option("spark.bigtable.project.id", projectId)
+  .option("spark.bigtable.instance.id", instanceId);
+```
+
+### Reading from Bigtable
+
+You can use the `spark.read()` format to read from Bigtable.
```

Comment:
similar to the above comment, maybe, With reading, you can use `bigtable` format, catalog, bigtable project id and instance to read from Bigtable.